11_RDS設定

shiny-a opened the pull request:
RDS設定ようやくできました。
パスワード等の機密情報はsecret.jsonの中に格納してgitignoreにて非公開設定を行なっています。
ですので、アプリ公開時には第三者から中身が分からないようになっています。

shiny-a committed:
RDS設定

Reimei1213 commented on config/settings.py:
```diff
@@ -11,13 +11,38 @@
 """
 
 from pathlib import Path
```
コンフリクト修正

Reimei1213 commented on config/settings.py:
```diff
@@ -76,8 +101,13 @@
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': BASE_DIR / 'db.sqlite3',
+        'ENGINE': 'django.db.backends.mysql',
+        'NAME': get_secret('DB_NAME'),
+        'USER': get_secret('DB_USER'),
+        'PASSWORD': get_secret('DB_PASSWORD'),
+        'HOST': 'rarestudy-db.cjs4w0auxrgb.us-east-2.rds.amazonaws.com',
+        'PORT': '3306',
+        'ATOMIC_REQUESTS': True,
```
コンフリクト修正

Reimei1213 reviewed (commented):
- [コンフリクト](https://backlog.com/ja/git-tutorial/pull-request/09/)とは競合が発生していることを指します。
- 作業ブランチにmainブランチをpullしたらコンフリクトが発生している旨のメッセージが表示され、該当箇所にどちらのブランチの内容を優先するか書いてあるので修正してください。

- importの部分はmainのものも作業ブランチのものも消さないでください。
- データベースの箇所は作業ブランチの方を優先してください。
- 修正が終わったら実際に動かしてみて問題がないか確認して、大丈夫でしたらプッシュした後修正した旨を教えてください。
- プルリクには[WIP]をつけてください。

Reimei1213 commented:
pip でインストールするmysqlclientのみrequirements.txtにバージョンを追記しておいてください。
```
mysqlclient==バージョン
```
https://github.com/shiny-a/RareStudy/blob/main/requirements.txt

Reimei1213 commented:
.gitignoreに追記していないっぽいので、ご確認お願いします。

shiny-a committed:
mysqlバージョン追加

shiny-a commented:
遅れてすみません。
追加しました。
.gitignoreではなくrequirements.txtに追加でよかったですか？

Reimei1213 commented:
言葉足らずで申し訳ございません。
requirements.txtの方はmysqlclientのバージョンを追記して欲しくて、gitignoreの方は更新されてなかったのでsecret.jsonが追記されていないと思いコメントしました。
あと、コンフリクトが解消していないようですがpullはしましたか？？

shiny-a committed:
コンフリクト修正

shiny-a commented:
今作業中です。。

shiny-a commented:
プッシュしました

shiny-a committed:
.gitignore secret.json追加

shiny-a committed:
Fix typo

MySQL Clientのバージョン追記の箇所が"MySQL"となっていたので修正しました。

shiny-a committed:
Module修正

DBに接続しようとしたところ、modelsディレクトリ内にあるファイル全ての"rarestudy.models.user"でエラー。
".user"ではなく"User"に修正したところDBに接続できました。
requirements.txtはMySQLの表記が小文字が正解だったぽいのでついでに修正してます。

shiny-a commented:
何度もすみません。。
修正ばかり気にして動作確認してませんでした。
詳細はプッシュした内容になります。

Reimei1213 committed:
Merge branch 'main' into test_shinya

Reimei1213 commented on requirements.txt:
```diff
@@ -1,2 +1,3 @@
 Django==3.2.3
+mysql-client==10.4.19-MariDB
```
mysql-clientではなくmysqlclientではないですかね？
また、バージョン10.4.19はないみたいなのでもう一度確認お願いします。
あと、-MariDBはおそらく不要です。

pip install -r requirements.txtを実行してみて大丈夫そうか確認お願いします。
(bootstrapのバージョンが間違っていたのでリモートのtest_shinyaをローカルにpullしておいてください)

shiny-a replied:
10.4.19はサーバーのバージョンでした。
勘違いしました。すみません。
改めて調べましたが、mysqlclient自体のバージョンはありません。
MySQLのバージョンがmysqlclientのバージョンになるようですが、どうしますか？

Reimei1213 replied:
mysqlclientにはバージョンはあります。
pip list or pip3 list と入力するとインストールしたパッケージ一覧があるので、そこからmysqlclientのバージョンを調べてください。

shiny-a replied:
ありました。
調べ方が悪かったですね。修正します。

Reimei1213 replied:
ありがとうございます。
mysqlclient==探したバージョンにしたら大丈夫だと思います。

shiny-a replied:
pip install -r requirements.txt正常に作動しました。

Reimei1213 committed:
Delete secrets.json

shiny-a committed:
mysqlclient修正

shiny-a committed:
import先のModule名修正(User→user)

shiny-a committed:
__init__.py Module名修正

Reimei1213 commented:
👍
